### PR TITLE
Position Score above fold

### PR DESCRIPTION
**Pull request by dfdumaresq**
As a newbie to the AIA, I found I was compelled to constantly scroll down to see how my choices had affected my Assessment Score. So I've moved the score panel to the middle-right of the screen.
The results are I can now see that my score can also influence the Impact level.
image

After adding Health Related services
image

### CODE REVIEW

**Comment by gcharest** [comment]
Oh, that's nice! I'm not in the office at the moment but will try to find a
minute to review!

**Comment by dfdumaresq** [comment]
Thanks, @gcharest!
The scores don't fit in the box when the locale is French. It's not pretty so I'm working on that now.
image

**Comment by gcharest** [comment]
OK, will wait before testing.

Also was wondering, what's the render like on mobile?

**Comment by dfdumaresq** [comment]
I think I've got this. A code review might be helpful though; this is my first time into Vue.

image

**Comment by dfdumaresq** [comment]
> Also was wondering, what's the render like on mobile?

I'll check and let you know, but this will from my localhost.

**Comment by dfdumaresq** [comment]
Ah yes. Visual Rendering (_not_ Rendering time... heh) It's not great, but better in Landscape mode:

IMG_2656

IMG_2654

**Comment by gcharest** [comment]
Thanks! Looking good in desktop mode in both locale!

OK, I'll check if there's a way to have the CSS behave differently in mobile in Vue. Not sure how, maybe @CalvinRodo or @e-wu would have an idea.

**Comment by dfdumaresq** [comment]
I'm thinking of using a navbar from Boostrap-vue. 
Here's a mockup of where I'm going.
image

**Comment by gcharest** [comment]
Oh, that's pretty nice!

Le dim. 11 août 2019 14 h 42, dfdumaresq <notifications@github.com> a
écrit :

**Comment by dfdumaresq** [comment]
I've implemented the above example. Styles could use a fair amount of tweaking.

image

**Comment by gcharest** [comment]
Cool! I don't have access to my computer today but will try to get to it this evening. Otherwise, I'm really not a designer so I'd probably try to touch base with @CalvinRodo who may know someone 🤔

**Comment by dfdumaresq** [comment]
Bootstrap-vue allowed me to remove the locale checking. That's a win!

image

**Comment by e-wu** [comment]
I like the bottom floating bar :) Nice job.

For later PRs that would be nice to have
- Can we add the "back" "forward" buttons as well? 
- Can the score be in initial "collapsed" state where it is maximum 1 liner, so when vertical resolution is small, the floating bar won't cover the viewport?

**Comment by MrDeshaies** [comment]
Great contribution, thank you Dave! What follows are a few minor niggles on design.

Using the hamburger/menu icon for collapsing doesn't seem right; a downward arrow would seem more appropriate, or the "-" like in Windows.

On the last screenshot (desktop version in French), the box is titled "AIA SCORE" which is English. Maybe we could simply use "Results" (English) and "Résultats" (French).

Finally, the color (heh). I'm not a huge fan of the teal. I suggest a light grey to harmonize with the other styles.

**Comment by dfdumaresq** [comment]
Thanks for the feedback guys! I’m going to be away from my computer for a few days, plus back to my day-job on Friday, so my productivity will scale back a bit. But I will address the issues you’re concerned with. 

A main concern is improving the dashboard for small phones in portrait mode so that it does not take up too much real estate. 
My challenge is positioning the four score categories. Is there any flexibility with the wording? 
I would like to try dropping the word score, which will give more room for the categories and values. I think the meaning will still be clear.

**Comment by dfdumaresq** [comment]
Wondering what you think of using icons in the score container? Here's an example with acronyms, which we could change to icons if in agreement.

Image-1

**Comment by dfdumaresq** [comment]
1) I've kept a single line container and I've dropped the Teal colour. 
2) I've added acronyms for the result labels on mobile screens. These are translated for French.

**Comment by gcharest** [comment]
That looks pretty good! What do you think @MrDeshaies ?

**Comment by MrDeshaies** [comment]
Yeeeessss! Looking good! Great job @gcharest @dfdumaresq

**Comment by gcharest** [comment]
All @dfdumaresq work! I will run a few accessibility tests this PM.

**Comment by dfdumaresq** [comment]
@gcharest, I've compared master with my branch and can see the accessibility has degraded. Input fields are narrower, select options are in a smaller font size, for example. I suspect my introduction of bootstrap is the culprit.
```"bootstrap": "^4.3.1", "bootstrap-vue": "^2.0.0-rc.28",``

**Comment by gcharest** [comment]
Thanks! Hmm, I'm think we might need to bring in @thriuin for input. Can we deploy this on open.canada.ca?

Who tests/vets the accesibility on your team? What would be your acceptance threshold?

Thanks!

**Comment by dfdumaresq** [comment]
@gcharest , I've fixed up the css issues. Have a look:

Desktop
aia-desktop-css-fix

Mobile - iPhone 6
aia-mobile-css-fix

**Comment by gcharest** [comment]
Awesome! I think we can merge now. I'll contact our colleagues at
open.canada.ca to confirm if they need anything else before deploying as
well!

Le sam. 7 sept. 2019 23 h 53, dfdumaresq <notifications@github.com> a
écrit :

**Comment by thriuin** [comment]
Hi @gcharest, back from leave. For deployment, I would start with the staging site first for review.

**Comment by gcharest** [comment]
Awesome, thanks for the feedback!

Le lun. 9 sept. 2019 10 h 46, Ross Thompson <notifications@github.com> a
écrit :

**Comment by gcharest** [comment]
Hi @dfdumaresq !

I was about to merge the PR but happened to be using Chrome for that and there seems to be an unexpected floating issue (I think?) Firefox doesn't create the same issue, the page width and its elements all fit right into place.

image

**Comment by dfdumaresq** [comment]
I'll take a look at it @gcharest.

**Comment by gcharest** [comment]
The last commit created the issue. I'm going over the CSS modifications to narrow down where it could have lead to this strange behaviour.

File: public/survey.css [hunk 1]
```diff
@@ -1,3 +1,168 @@
...
+}
+
+.container-fluid {
+  padding-right: 15px !important;
+  padding-left: 15px !important;
+  margin-right: auto !important;
+  margin-left: auto !important;
+}
+
+.h1, .h2, .h3, .h4, .h5, .h6, h1, h2, h3, h4, h5, h6 {
```
Comment (gcharest) [inline public/survey.css]: Issues seems wot bhe with the changes to headers somehow.

Comment (dfdumaresq) [reply public/survey.css]: Okay - I'm glad you've resolved the floating issue.

Comment (gcharest) [reply public/survey.css]: Was rushing to get the kids, haven't had the chance to fix it yet lol :grin:

Comment (dfdumaresq) [reply public/survey.css]: Ah - I have one too (he just turned 5 and started Kindergarten!) I thought you wanted me to fix this... So I'll get out of your way. Just let me know if you want me to do anything.

Comment (gcharest) [reply public/survey.css]: No, no, if you have a few cycles, you're way better than me at CSS just from looking at the magic you've done :grinning:

Comment (gcharest) [reply public/survey.css]: So, as far as I can tell, the issue is with h2, which seems to be used throught the [WET](https://www.canada.ca/en/treasury-board-secretariat/services/government-communications/web-experience-toolkit.html) template via the script for screen reader content.

By changing the behaviour of the CSS for our own h2 headers, it looks like it interferes with the injected HTML. At least that's what I think is happening :thinking:

Comment (gcharest) [reply public/survey.css]: My only workaround for now would be to remove h2 as an element to be included on line 32.

If you have another option, please go ahead. If not, if you could please fix and push your commit, I'll merge the PR and we'll have a new UI for the score! :grinning: 

Thanks so much for all the great work you've done on this project!

Comment (dfdumaresq) [reply public/survey.css]: Okay - I'm on it.

Comment (dfdumaresq) [reply public/survey.css]: The only problem is I can't reproduce what you're seeing. 
I'm using Chrome Version 77. This is what I get when I hit the 'Start Again' button.

image

Comment (dfdumaresq) [reply public/survey.css]: Your image above seems to suggest you have scrollbars appearing in the window. Could you try Window|Zoom? See if that makes a difference? Also, of course clear your cache. You can also try a Empty Cache and Hard Reload, but you need the Inspector open to see that after holding down the reload button.

image

Comment (dfdumaresq) [reply public/survey.css]: Safari v13
image

Firefox v69
image

Comment (gcharest) [reply public/survey.css]: OK, testing it out!

Comment (gcharest) [reply public/survey.css]: Hmm, so I tested on Chrome 77, both on Windows 10 Pro and Ubuntu 18.04 and both behave the same. The bottom scroll bar is there for Chrome but not other browsers 🤔 

Behaviour is as expected with Edge 44 and Firefox 69.0.1. This is really odd... Will try on a MacOS device to see what's going on.

Comment (gcharest) [reply public/survey.css]: image

Comment (dfdumaresq) [reply public/survey.css]: I can now reproduce the scrollbars you mention in Chrome, I think. But tell me are you able to scroll to the left and show the page correctly?

Comment (dfdumaresq) [reply public/survey.css]: I've followed your suggestion and it works on my Mac.

image

**Comment by gcharest** [comment]
Thanks so much! This is really great work!

**Comment by dfdumaresq** [comment]
Excellent, @gcharest! That was fun!
I've found another grammar correction and I'll look for some other issues I can tackle. Let me know if there's anything in particular that you'd like me to look at.

**Pull request merged**